### DISCUSSION TITLE
not able to connect to mosquitto mqtt since latest update

### DISCUSSION THREAD

**Discussion by wilcovh** [teslamate-org/teslamate#1965] [Q&A]
running teslamate in docker on raspberry pi. telsamate itself works fine but it seems that since last update the mqtt server can not be reached anymore. Have been using node-red on another device to connect to teslamate mqtt server to grab data but that does not connect anymore. Has something changed that I have missed?

**Comment by wilcovh** [comment]
mosquitto has updated their security. so check the docker-compose.yml it needs updated with one line under mosquitto

**Reply by eliluong** [reply]
Can you provide more details on what this one-line update is? Will be useful for future reference.

**Reply by wilcovh** [reply]
command: mosquitto -c /mosquitto-no-auth.conf

**Reply by grw000** [reply]
I appear to be having the same problem.  I have tried adding the line"command: mosquitto -c /mosquitto-no-auth.conf" and restarted teslamate, but this doesn't seem to have fixed the problem.  Any advice and guidance would be appreciated.

**Reply by wilcovh** [reply]
your` docker-compose.yml looks like this?

mosquitto:
    image: eclipse-mosquitto:2
    restart: always
    command: mosquitto -c /mosquitto-no-auth.conf
    ports:
      - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data
`

**Reply by wilcovh** [reply]
this is how i got it working>

**Reply by grw000** [reply]
Apologies for the follow up question ... I am still having trouble getting the mosquitto working.  I have updated my docker-compose.yml file as suggested, but the logs indicate that something is still wrong (see below).  Any suggestions on how to investigate and resolve this would be greatly appreciated.

pi@teslamate:~/tesla $ docker-compose logs teslamate | more
Attaching to tesla_teslamate_1
teslamate_1  | 2021-09-13 23:18:37.208 [info] Migrations already up
teslamate_1  | 2021-09-13 23:18:44.841 [info] System Info: Erlang/OTP 24 (emu)
teslamate_1  | 2021-09-13 23:18:44.842 [info] Version: 1.24.0
teslamate_1  | 2021-09-13 23:18:45.683 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (741.790 ms)
teslamate_1  | 2021-09-13 23:18:46.458 [info] POST https://owner-api.teslamotors.com/oauth/token -> 200 (773.794 ms)
teslamate_1  | 2021-09-13 23:18:46.458 [info] Refreshed api tokens
teslamate_1  | 2021-09-13 23:18:46.476 [info] Scheduling token refresh in 5 wk 5 d
teslamate_1  | 2021-09-13 23:18:46.486 [info] Running TeslaMateWeb.Endpoint with cowboy 2.9.0 at :::4000 (http)
teslamate_1  | 2021-09-13 23:18:46.490 [info] Access TeslaMateWeb.Endpoint at http://localhost
teslamate_1  | 2021-09-13 23:18:46.706 [info] Starting logger for 'Gerald'
teslamate_1  | 2021-09-13 23:18:47.057 car_id=1 [info] Start / :asleep
teslamate_1  | 2021-09-13 23:18:49.223 [info] tzdata release in place is from a file last modified Tue, 22 Dec 2020 23:35:21 GMT. Release file 
on server was last modified Sun, 24 Jan 2021 19:35:23 GMT.
teslamate_1  | 2021-09-13 23:18:49.417 [info] GET /geo-fences/4/edit
teslamate_1  | 2021-09-13 23:18:49.448 [info] Sent 200 in 30ms
teslamate_1  | 2021-09-13 23:18:49.869 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3639313"}} terminating
teslamate_1  | ** (stop) :ehostunreach
teslamate_1  | Last message: :connect
teslamate_1  | 2021-09-13 23:18:52.067 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.068 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.068 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.069 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.069 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.070 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.070 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.070 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.071 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.071 [warn] MQTT publishing failed: :timeout
teslamate_1  | 2021-09-13 23:18:52.989 [error] GenServer {Tortoise.Registry, {Tortoise.Connection, "TESLAMATE_302E3639313"}} terminating
teslamate_1  | ** (stop) :ehostunreach

**Reply by wilcovh** [reply]
is the mosquitto docker container running?

**Reply by grw000** [reply]
Thanks again for your help.  I ran a "docker-compose ps" command and it looks like mosquitto is having problems being started:

      Name                     Command                 State                       Ports                  
----------------------------------------------------------------------------------------------------------
tesla_database_1    docker-entrypoint.sh postgres    Up           5432/tcp                                
tesla_grafana_1     /run.sh                          Up           0.0.0.0:3000->3000/tcp,:::3000->3000/tcp
tesla_mosquitto_1   /docker-entrypoint.sh mosq ...   Restarting                                           
tesla_teslamate_1   tini -- /bin/sh /entrypoin ...   Up           0.0.0.0:4000->4000/tcp,:::4000->4000/tcp

I notice from my DNS (running on another Raspberry Pi), that my teslamate server is continually trying to resolve the hostname "mosquitto".  I've added a DNS entry to resolve "mosquitto" to the IP address of the teslamate host but this doesn't appear to have any impact.

It would seem like I have some sort of TCP/IP configuration problem but I am having problems figuring out what is wrong.

Thanks again.
